File: examples/javascript/basic.mjs
// Basic Veil AI Firewall + OpenAI example.
import OpenAI from 'openai';
import { createVeilOpenAIConfig } from 'a5omic-veil';

const openai = new OpenAI(createVeilOpenAIConfig({
  veilApiKey: process.env.VEIL_API_KEY,
  upstreamApiKey: process.env.OPENAI_API_KEY,
  inputPolicy: 'block',
  outputPolicy: 'monitor',
}));

const response = await openai.chat.completions.create({
  model: 'gpt-4o-mini',
  messages: [{
    role: 'user',
    content: `Summarize this ticket:
From: Jane Doe (jane@company.com)
Phone: 555-123-4567
SSN: 123-45-6789
Issue: Double charged $299.99 for Premium plan.`
  }]
});

console.log(response.choices[0].message.content);

File: examples/python/basic.py
"""Basic Veil AI Firewall + OpenAI example."""
import os
from openai import OpenAI

client = OpenAI(
    api_key=os.environ["OPENAI_API_KEY"],
    base_url="https://veil-api.com/v1",
    default_headers={
        "Authorization": f"Bearer {os.environ['VEIL_API_KEY']}",
        "x-upstream-key": os.environ["OPENAI_API_KEY"],
        "x-veil-input-policy": "block",
        "x-veil-output-policy": "monitor",
    },
)

response = client.chat.completions.create(
    model="gpt-4o-mini",
    messages=[
        {
            "role": "user",
            "content": (
                "Summarize this support ticket:\n\n"
                "From: Sarah Johnson (sarah.j@acme.com)\n"
                "Phone: 408-555-7721\n"
                "SSN: 521-03-8847\n"
                "Card: 4111111111111111\n\n"
                "Issue: Charged twice for Premium subscription."
            ),
        }
    ],
)

print(response.choices[0].message.content)
